File: src/main/java/ru/kantser/elephantmusic/model/AppSettings.java
package ru.kantser.elephantmusic.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;

public class AppSettings {
    @JsonProperty("language")
    @Getter @Setter
    private String language;
    
    @JsonProperty("lastFmName")
    @Getter @Setter
    private String lastFmName;
    
    @JsonProperty("lastFmToken")
    @Getter @Setter
    private String lastFmToken;

    @JsonProperty("activeScrobbling")
    @Getter @Setter
    private Boolean activeScrobbling;

    @JsonProperty("updateSourceUrl")
    @Getter @Setter
    private String updateSourceUrl;

    @JsonProperty("updateToken")
    @Getter @Setter
    private String updateToken;

}

File: compose-playground/src/commonMain/kotlin/ru/kantser/elephantmusic/domain/model/AppSettings.kt
package ru.kantser.elephantmusic.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class AppSettings(
    val language: String = "RU",
    val lastFmName: String = "Anonim",
    val lastFmToken: String = "NULL",
    val activeScrobbling: Boolean = true,
    val volume: Double = 0.8,
    val updateSourceUrl: String? = null,
    val updateToken: String? = null,
    val auddToken: String? = null,
    val acrAccessKey: String? = null,
    val acrAccessSecret: String? = null,
    val acrHost: String? = null,
)
